js/fairdesk.js

```diff
@@ -0,0 +1,1450 @@
+    }
+
+    parseTicker (ticker, market = undefined) {
+        const marketId = this.safeString (ticker, 'symbol');
+        market = this.safeMarket (marketId, market);
+        const symbol = market['symbol'];
+        const timestamp = this.safeInteger (ticker, 'timestamp');
+        const last = this.safeNumber (ticker, 'close');
+        const open = this.safeNumber (ticker, 'open');
+        const change = last - open;
+        const percentage = ((last - open) / open) * 100;
+        return this.safeTicker ({
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeString (ticker, 'high'),
+            'low': this.safeString (ticker, 'low'),
+            'open': open.toString (),
+            'close': last.toString (),
+            'last': last.toString (),
+            'bid': undefined,
+            'bidVolume': undefined,
+            'ask': undefined,
+            'askVolume': undefined,
+            'vwap': undefined,
+            'previousClose': undefined, // previous day close
+            'change': change.toString (),
+            'percentage': percentage.toString (),
+            'average': this.safeString (ticker, 'averagePrice'),
+            'baseVolume': this.safeString (ticker, 'baseVolume'),
+            'quoteVolume': this.safeString (ticker, 'quoteVolume'),
+            'info': ticker,
+        }, market, false);
+    }
+
+    parseSymbol (symbol) {
+        const str = symbol.replace ('/', '');
+        return str.toLowerCase ();
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        const symbolId = this.parseSymbol (symbol);
+        const market = this.market (symbolId);
+        const request = {
+            'symbol': market['id'],
+        };
+        const response = await this.publicGetMdTicker24h (this.extend (request, params));
+        // {
+        //     "status": 0,
+        //     "error": "OK",
+        //     "data": {
+        //       "symbol": "btcusdt",
+        //       "timestamp": 1651895571086,
+        //       "open": 36439.500000,
+        //       "high": 36542.500000,
+        //       "low": 35344.500000,
+        //       "close": 35890.000000,
+        //       "indexPrice": 35891.0,
+        //       "markPrice": 35892.9,
+        //       "openInterest": 74.275366,
+        //       "fundingRate": 0.00010000,
+        //       "predicateFundingRate": 0.000100,
+        //       "baseVolume": 313.214000,
+        //       "quoteVolume": 11280153.148
+        //     }
+        // }
+        const result = this.safeValue (response, 'data', {});
+        return this.parseTicker (result, market);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        const symbolId = this.parseSymbol (symbol);
+        await this.loadMarkets ();
+        const market = this.market (symbolId);
```

Comment:
We must pass the original `symbol` arg to `this.market`, for it to work market ids have to be filled in `fetchMarkets()`.